src/services/logqlMatchers.ts

```diff
@@ -167,18 +209,133 @@ function parseLineFilters(query: string, lineFilters: LineFilterType[]) {
   }
 }
 
-export function getMatcherFromQuery(query: string): { labelFilters: Filter[]; lineFilters?: LineFilterType[] } {
-  const filter: Filter[] = [];
+function getNumericFieldOperator(matcher: SyntaxNode) {
+  if (getAllPositionsInNodeByType(matcher, Lte).length) {
+    return FilterOperator.lte;
+  } else if (getAllPositionsInNodeByType(matcher, Lss).length) {
+    return FilterOperator.lt;
+  } else if (getAllPositionsInNodeByType(matcher, Gte).length) {
+    return FilterOperator.gte;
+  } else if (getAllPositionsInNodeByType(matcher, Gtr).length) {
+    return FilterOperator.gt;
+  }
+
+  console.warn('unknown numeric operator');
+
+  return undefined;
+}
+
+function getStringFieldOperator(matcher: SyntaxNode) {
+  if (getAllPositionsInNodeByType(matcher, Eq).length) {
+    return FilterOperator.Equal; // =
+  } else if (getAllPositionsInNodeByType(matcher, Neq).length) {
+    return FilterOperator.NotEqual; // !=
+  } else if (getAllPositionsInNodeByType(matcher, Re).length) {
+    console.warn('field regex not currently supported'); // =~
+  } else if (getAllPositionsInNodeByType(matcher, Nre).length) {
+    console.warn('field exclusive regex not currently supported'); // !~
+  }
+
+  return undefined;
+}
+
+function parseFields(query: string, fields: FieldFilter[], context: PluginExtensionPanelContext, lokiQuery: LokiQuery) {
+  const dataFrame = context.data?.series.find((frame) => frame.refId === lokiQuery.refId);
+  // We do not currently support "or" in Explore logs, so grab the left hand side LabelFilter leaf nodes as this will be the first filter expression in a given pipeline stage
+  const allFields = getLHSLeafNodesFromQuery(query, [LabelFilter]);
+
+  for (const matcher of allFields) {
+    const position = NodePosition.fromNode(matcher);
+    const expression = position.getExpression(query);
+
+    // Skip error expression, it will get added automatically when explore logs adds a parser
+    if (expression.substring(0, 9) === `__error__`) {
+      continue;
+    }
+
+    // @todo we need to use detected_fields API to get the "right" parser for a specific field
+    // Currently we just check to see if there is a parser before the current node, this means that queries that are placing metadata filters after the parser will query the metadata field as a parsed field, which will lead to degraded performance
```

Comment:
I guess in the app we could check if a field is actually metadata once we execute the detected_fields call and move the filter from the fields variable to the metadata variable?